src/ArrayManifold.jl

```diff
@@ -218,6 +218,29 @@ function vector_transport_to!(
     return vto
 end
 
+function vector_transport_to!(
+    M::ArrayManifold,
+    vto,
+    x,
+    v,
+    y,
+    m::ProjectionTransport;
+    kwargs...,
+)
+    is_manifold_point(M, y, true; kwargs...)
+    is_tangent_vector(M, x, v, true; kwargs...)
+    vector_transport_to!(
+        M.manifold,
+        array_value(vto),
+        array_value(x),
+        array_value(v),
+        array_value(y),
+        m,
+    )
+    is_tangent_vector(M, y, vto, true; kwargs...)
+    return vto
+end
```

Comment:
An and even, this also adds the tests that we document `ArrayManifold` does (but did not do until now), so also semantically I think this is a good idea.